### PR TITLE
Average function

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented weighted average function. It is compatible with `numpy.average`.
We can reproduce a reduced version of loss function with this function and an elementwise-loss function.

### CODE REVIEW

**Comment by unnonouno** [comment]
related to #2357

File: chainer/functions/math/average.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+def average(x, axis=None, weights=None):
+    """Calculate weighted average of array elements over a given axis.
+
+    Args:
+        x (~chainer.Variable): Elements to sum.
+        axis (None or int): Axis which the method is performed.
+            The default (axis = None) is perform a mean over all the dimensions
+            of the input array.
+        weights (None or ~chainer.Variable): An array holding weights to
```
Comment (okuta) [inline chainer/functions/math/average.py]: ditto

File: chainer/functions/math/average.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from chainer.functions.array import broadcast
+from chainer.functions.array import reshape
+from chainer.functions.math import sum as sum_mod
+
+
+def average(x, axis=None, weights=None):
+    """Calculate weighted average of array elements over a given axis.
+
+    Args:
+        x (~chainer.Variable): Elements to sum.
```
Comment (unnonouno) [inline chainer/functions/math/average.py]: Yes, but i didn't write test yet. We also need to refactor all other tests.

Comment (okuta) [inline chainer/functions/math/average.py]: OK.

**Comment by okuta** [comment]
OK.
Please add this function to rst file.

**Comment by unnonouno** [comment]
I fixed it.

**Comment by okuta** [comment]
OK.

**Pull request merged**